renderapi/client.py

```diff
@@ -859,3 +860,192 @@ def transformSectionClient(stack, transformId, transformClass, transformData,
+                     excludeFirstTransformAndAllAfter=None,    
+                     subprocess_mode=None,
+                     host=None, port=None,
+                     owner=None, project=None, client_script=None,
+                     memGB=None, render=None, **kwargs):
+    """run SiftPointMatchClient.java
+
+    Parameters
+    ----------
+    stack : str
+        stack containing the tiles
+    collection : str
+        point match collection to save results into
+    tile_pairs : iterable
+        list of iterables of length 2 containing tileIds to calculate point matches between
+    sift_options: SiftOptions
+        options for running point matching
+    pointMatchRender : renderapi.render.renderaccess
+        renderaccess object specifying the render server to store point matches in
+        defaults to values specified by render and its keyword argument overrides
+    debugDirectory : str
+        directory to store debug results (optional)
+    filter: bool
+        whether to apply default filtering to tile (default=False)
+    renderWithoutMask: bool
+        whether to exclude the mask when rendering tile (default=False)
+    normalizeForMatching: bool
+        whether to apply traditional 'normalizeForMatching' transform manipulation to image
+        this removes the last transform from the transformList, then if there are more than 3 transforms
+        continues to remove transforms until there are exactly 3.  Then assumes the image will be near 0,0
+        with a width/height that is about equal to the raw image width/height.  This is true for Janelia's
+        conventions for transformation alignment, but use at your own risk. (default=True)
+    excludeTransformsAfterLast: str or None
+        alternative to normalizeForMatching, which uses transformLabels.  Will remove all transformations
+        after the last transformation with this transform label.  i.e. if all lens corrections have a 'lens'
+        label.  Then this will remove all non-lens transformations from the list.
+        This is more general than normalizeForMatching=true, but requires you have transform labels applied.
+        default = None
+    excludeFirstTransformAndAllAfter: str
+        alternative to normalizeForMatching which finds the first transform in the list with a given label
+        and then removes that transform and all transforms that follow it. i.e. if you had a compound list
+        of transformations, and you had labelled the first non-local transform 'montage' then setting
+        excludeFirstTransformAndAllAfter='montage' would remove that montage transform and any other
+        transforms that you had applied after it. default= None.
+    excludeAllTransforms: bool
+        alternative to normalizeForMatching which simply removes all transforms from the list.
+        default=False
+
+    """
+    if pointMatchRender is None:
+        pointMatchRender = Render(host, port, owner, project, client_script)

```

Comment:
This, I think, does not allow us to overwrite with specified host, port, owner parameters if pointMatchRender is not None.  because this is not related to a regular render option it is not a huge departure from the way the rest of the module works, but still a little counterintuitive.

Reply:
hmm.. good point... but this is the only cross tilespec, point match api call in one call.... so some possibilities.  1) don't allow overwriting at all here.. have tilespecRender and pointmatchRender (optionally) 2) have host and pm_host, port and pm_port, etc...  3) ??